org.eclipse.scout.rt.dataobject/src/main/java/org/eclipse/scout/rt/dataobject/id/IdCodec.java

```diff
@@ -248,17 +349,60 @@ protected IId fromQualifiedInternal(String qualifiedId, boolean lenient) {
    * @throws PlatformException
    *           if the given string does not match the expected format
    */
-  protected <ID extends IId> ID fromUnqualifiedUnchecked(Class<ID> idClass, String unqualifiedId) {
+  protected <ID extends IId> ID fromUnqualifiedUnchecked(Class<ID> idClass, String unqualifiedId, Set<IIdCodecFlag> flags) {
+    unqualifiedId = decryptUnqualified(idClass, unqualifiedId, flags);
     String[] rawComponents = unqualifiedId.split(";", -1 /* force empty strings for empty components */);
-    Object[] components = parseComponents(idClass, rawComponents);
-    return m_idFactory.get().createInternal(idClass, components);
+    Object[] components = parseComponents(idClass, rawComponents, flags);
+    return idFactory().createInternal(idClass, components);
+  }
+
+  /**
+   * Decrypts the given unqualifiedId iff
```

Comment:
javadoc is a little missleading: this method per se parses the unqualified id and eventually checks signature iff ....